python-sdk/rebuff/rebuff_pythonsdk.py

```diff
@@ -0,0 +1,228 @@
+import secrets
+from typing import Any, Optional, Tuple, Union
+from detect_pi_heuristics import detect_prompt_injection_using_heuristic_on_input
+from detect_pi_vectorbase import init_pinecone, detect_pi_using_vector_database
+from detect_pi_openai import render_prompt_for_pi_detection, call_openai_to_detect_pi
+from pydantic import BaseModel
+import langchain
+
+
+class Rebuff_Detection_Response(BaseModel):
+    heuristic_score: float
+    openai_score: float
+    vector_score: float
+    run_heuristic_check: bool
+    run_vector_check: bool
+    run_language_model_check: bool
+    max_heuristic_score: float
+    max_model_score: float
+    max_vector_score: float
+    injection_detected: bool
+
+
+class RebuffPython:
+    def __init__(
+        self,
+        openai_model: str,
+        openai_apikey: str,
+        pinecone_apikey: str,
+        pinecone_environment: str,
+        pinecone_index: str,
+    ) -> None:
+        self.openai_model = openai_model
+        self.openai_apikey = openai_apikey
+        self.pinecone_apikey = pinecone_apikey
+        self.pinecone_environment = pinecone_environment
+        self.pinecone_index = pinecone_index
+
+    def detect_injection(
+        self,
+        user_input: str,
+        max_heuristic_score: float = 0.75,
+        max_vector_score: float = 0.90,
+        max_model_score: float = 0.90,
+        check_heuristic: bool = True,
+        check_vector: bool = True,
+        check_llm: bool = True,
+    ) -> Rebuff_Detection_Response:
+        """
+        Detects if the given user input contains an injection attempt.
+
+        Args:
+            user_input (str): The user input to be checked for injection.
+            max_heuristic_score (float, optional): The maximum heuristic score allowed. Defaults to 0.75.
+            max_vector_score (float, optional): The maximum vector score allowed. Defaults to 0.90.
+            max_model_score (float, optional): The maximum model (LLM) score allowed. Defaults to 0.9.
+            check_heuristic (bool, optional): Whether to run the heuristic check. Defaults to True.
+            check_vector (bool, optional): Whether to run the vector check. Defaults to True.
+            check_llm (bool, optional): Whether to run the language model check. Defaults to True.
+
+        Returns:
+            Rebuff_Detection_Response
+        """
+
+        injection_detected = False
+
+        if check_heuristic:
+            rebuff_heuristic_score = detect_prompt_injection_using_heuristic_on_input(
+                user_input
+            )
+
+        else:
+            rebuff_heuristic_score = 0
+
+        if check_vector:
+            vector_store = init_pinecone(
```

Comment:
I'm not sure how much performance would be gained, but you could call `init_pinecone` once and save the `vector_store` in an instance attribute. That's how we do it in the JavaScript SDK.

Reply:
Thank you for the suggestion, I have now added `vector_store` as an instance attribute.

Reply:
I'd recommend putting:
```py
        if self.vector_store is None:
            self.vector_store = init_pinecone(
                self.pinecone_environment,
                self.pinecone_apikey,
                self.pinecone_index,
                self.openai_apikey,
            )
```
into a helper function.